ocean_lib/assets/v4/asset.py

```diff
@@ -0,0 +1,264 @@
+            == trusted_algorithms
+        ), "New trusted algorithm was not removed. Failed when updating the list of trusted algorithms. "
+
+        return trusted_algorithms
+
+    def remove_publisher_trusted_algorithm_publisher(
+        self,
+        compute_service: V4Service,
+        publisher_address: str,
+    ) -> list:
+        """
+        :return: List of trusted algo publishers not containing `publisher_address`.
+        """
+
+        trusted_algorithm_publishers = [
+            tp.lower() for tp in compute_service.get_trusted_algorithm_publishers()
+        ]
+        publisher_address = publisher_address.lower()
+        if not trusted_algorithm_publishers:
+            raise ValueError(
+                f"Publisher {publisher_address} is not in trusted algorithm publishers of this asset."
+            )
+
+        trusted_algorithm_publishers = [
+            tp for tp in trusted_algorithm_publishers if tp != publisher_address
+        ]
+        trusted_algorithms = compute_service.get_trusted_algorithms()
+        self.update_compute_values(
+            trusted_algorithms, trusted_algorithm_publishers, True, False
+        )
+        assert (
+            self.get_service("compute").compute_values[
+                "publisherTrustedAlgorithmPublishers"
+            ]
+            == trusted_algorithm_publishers
+        ), "New trusted algorithm publisher was not removed. Failed when updating the list of trusted algo publishers. "
+
+        return trusted_algorithm_publishers
+
+    def update_compute_values(
+        self,
+        trusted_algorithms: List,
+        trusted_algo_publishers: Optional[List],
+        allow_network_access: bool,
+        allow_raw_algorithm: bool,
+    ) -> None:
+        """Set the `trusted_algorithms` on the compute service.
+
+        - An assertion is raised if this asset has no compute service
+        - Updates the compute service in place
+        - Adds the trusted algorithms under privacy.publisherTrustedAlgorithms
+
+        :param trusted_algorithms: list of dicts, each dict contain the keys
+            ("containerSectionChecksum", "filesChecksum", "did")
+        :param trusted_algo_publishers: list of strings, addresses of trusted publishers
+        :param allow_network_access: bool -- set to True to allow network access to all the algorithms that belong to this dataset
+        :param allow_raw_algorithm: bool -- determine whether raw algorithms (i.e. unpublished) can be run on this dataset
+        :return: None
+        :raises AssertionError if this asset has no `ServiceTypes.CLOUD_COMPUTE` service
+        """
+        assert not trusted_algorithms or isinstance(trusted_algorithms, list)
+        service = self.get_service(ServiceTypesV4.CLOUD_COMPUTE)
+        assert service is not None, "this asset does not have a compute service."
+
+        trusted_algorithms = trusted_algorithms if trusted_algorithms else []
+        if trusted_algorithms:
```

Comment:
no need for this if. If the list is empty, "for" will iterate an empty list and not do anything.